aten/src/ATen/native/FractionalMaxPool2d.cpp

```diff
@@ -0,0 +1,363 @@
+#include "ATen/ATen.h"
+#include "ATen/NativeFunctions.h"
+
+#include <tuple>
+#include <vector>
+
+namespace at {
+namespace native {
+namespace {
+
+template <typename scalar_t>
+static std::vector<int> fractional_max_pool2d_generate_intervals(
+  scalar_t sample,
+  int inputSize,
+  int outputSize,
+  int poolSize) {
+  scalar_t alpha = (scalar_t) (inputSize - poolSize) / (scalar_t) (outputSize - 1);
+  std::vector<int> sequence(outputSize, 0);
```

Comment:
Nice!

nit: But you immediately fill the elements afterward so there is no need to zero-fill it initially. (If you really want to go to hay on this, you don't even need explicit indexing anymore; you could initialize sequence zero-length, reserve and then emplace elements into it.)